base/acme/src/main/java/org/dogtagpki/acme/database/LDAPDatabase.java

```diff
@@ -432,11 +430,12 @@ private static ACMEOrder loadOrder(LDAPEntry entry)
         }
 
         List<ACMEIdentifier> identifiers = new ArrayList<>();
-        for (String type : IDENTIFIER_TYPES) {
-            attr = entry.getAttribute(ATTR_IDENTIFIER + ";" + type);
-            for (String value : Collections.list(attr.getStringValues())) {
-                identifiers.add(new ACMEIdentifier(type, value));
-            }
+        attr = entry.getAttribute(ATTR_IDENTIFIER);
+        for (String identifier : attr.getStringValueArray()) {
+            String[] parts = StringUtils.split(identifier, ":", 2);
+            String type = parts[0];
+            String value = parts[1];
```

Comment:
Here let us handle the case where ':' is not present in the string.  Both to handle old data, but more importantly to be robust in case of incorrect/corrupt data in database somehow.  Better to handle gracefully (meaningful error or perhaps just skip the value) than fail with index-out-of-bounds exception.

Whether we provide a fallback to handle the original scheme that used attribute description option I don't mind either way, but it would be nice.

Reply:
OK, I added a code to throw an exception if the identifier has an incorrect format, but I think this code will only read `acmeIdentifier`, not `acmeIdentifier;dns`. If we want to read the old attribute we probably would have to keep the old code.

I think going forward we should focus on upgrading the database instead of getting stuck with an old database. There's a plan to implement a [database upgrade framework](https://pagure.io/dogtagpki/issue/710) in PKI 10.10, but I don't know how much will actually get implemented. Hopefully we can use that to migrate ACME database.